assets-sync.js

```diff
@@ -1407,15 +1407,15 @@ class WorldMigration {
                 return localPath;
             }
         } else {
-            const asset = this.assets.get(decodeURI(name));
+            const asset = this.assets.get(ForgeAssetSync.sanitizePath(decodeURIComponent(name)));
             const queryString = query ? `?${query}` : "";
             // Same path, not bazaar and same url.. so it's not coming from someone else's library
             if (asset && `${asset.url}${queryString}` === entityPath) {
-                return `${name}${queryString}`;
+                return `${asset.name}${queryString}`;
             }
             // Wildcards will never work through https and can't be found in the Map, so we might as well just replace them as is
             if (supportsWildcard && name.includes("*")) {
-                return `${name}${queryString}`;
+                return `${ForgeAssetSync.sanitizePath(decodeURIComponent(name))}${queryString}`;
```

Comment:
We were not able to find the matching local asset, because of the wildcard. So we return the sanitized path that matches what the assets would have been synced to, rather than the URI encoded unsanitized path, which might differ if special characters were involved.